plugins/MauticFullContactBundle/EventListener/LeadSubscriber.php

```diff
@@ -0,0 +1,160 @@
+            $fullcontact = new FullContact_Person($keys['apikey']);
+            try {
+                /** @var User $user */
+                $user      = $this->userHelper->getUser();
+                $webhookId = 'fullcontact_notify#'.$lead->getId().'#'.$user->getId();
+                $cache     = $lead->getSocialCache() ?: [];
+                $cacheId   = sprintf('%s%s', $webhookId, date(DATE_ATOM));
+                if (!array_key_exists($cacheId, $cache)) {
+                    $fullcontact->setWebhookUrl(
+                        $this->router->generate(
+                            'mautic_plugin_fullcontact_index',
+                            [],
+                            UrlGeneratorInterface::ABSOLUTE_URL
+                        ),
+                        $webhookId
+                    );
+                    $res             = $fullcontact->lookupByEmailMD5(md5($lead->getEmail()));
+                    $cache[$cacheId] = serialize($res);
+                    $lead->setSocialCache($cache);
+                    $this->leadModel->saveEntity($lead);
+                }
+            } catch (\Exception $ex) {
+                $this->logger->log('error', 'Error while using FullContact: '.$ex->getMessage());
+            }
+        }
+    }
+
+    public function companyPostSave(CompanyEvent $event)
+    {
+        /** @var FullContactIntegration $myIntegration */
+        $myIntegration = $this->integrationHelper->getIntegrationObject('FullContact');
+
+        if (!$myIntegration->getIntegrationSettings()->getIsPublished()) {
+            return;
+        }
+
+        $company = $event->getCompany();
+
+        // get api_key from plugin settings
+        $keys = $myIntegration->getDecryptedApiKeys();
+
+        if ($myIntegration->shouldAutoUpdate()) {
+            $fullcontact = new FullContact_Company($keys['apikey']);
+            try {
+                /** @var User $user */
+                $user      = $this->userHelper->getUser();
+                $webhookId = 'fullcontactcomp_notify#'.$company->getId().'#'.$user->getId();
+                $parse     = parse_url($company->getFieldValue('companywebsite', 'core'));
+                $cache     = $company->getSocialCache() ?: [];
+                $cacheId   = sprintf('%s%s', $webhookId, date(DATE_ATOM));
+                if (!array_key_exists($cacheId, $cache)) {
+                    $fullcontact->setWebhookUrl(
+                        $this->router->generate(
+                            'mautic_plugin_fullcontact_index',
+                            [],
+                            UrlGeneratorInterface::ABSOLUTE_URL
+                        ),
+                        $webhookId
+                    );
+                    $res             = $fullcontact->lookupByDomain($parse['host']);
+                    $cache[$cacheId] = serialize($res);
+                    $company->setSocialCache($cache);
+                    $this->companyModel->saveEntity($company);
```

Comment:
Same thing here as Lead. Use `$this->companyModel->getRepository()->saveEntity` to prevent a event loop.